src/app/search-page/searchPage.controller.js

```diff
@@ -52,6 +58,16 @@ export default class SearchPageController {
                 //Do nothing on dialog cancel
             });
     }
+
+    mapStateToThis(state) {
+        return {
+            latestSearches: state.main.cities
+        };
+    }
+
+    $onDestroy() {
```

Comment:
I would put the $onDestroy close to the $onInit, in order to have all the angular events together